.github/workflows/CI-distro.yml

```
@@ -0,0 +1,75 @@
+name: Test GAP package distro
+
+on:
+  pull_request:
+  schedule:
+    # Every Monday at 3:08 AM UTC
+    - cron: '8 3 * * 1'
+  workflow_dispatch:
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the default repository branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref_name != github.event.repository.default_branch || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      gap-packages: ${{ steps.set-matrix.outputs.gap-packages }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: '1'
+      - name: "Cache artifacts"
+        uses: julia-actions/cache@v2
+      - name: "Build package"
+        uses: julia-actions/julia-buildpkg@v1
+      - name: Get list of GAP packages
+        id: set-matrix
+        run: julia --project=. -e '
+                  using Artifacts;
+                  output = sprint(print, "gap-packages=", filter(name -> startswith("d", name) || startswith("f", name), readdir(artifact"gap_packages")));
```

```suggestion
                  output = sprint(print, "gap-packages=", readdir(artifact"gap_packages"));
```
once we are happy with how this works, we can enable it to test the whole distro and not just ~10 packages